### ISSUE TITLE
Materialize functions can not be found

### ISSUE THREAD

**Issue by leNicDev** [InfomediaLtd/angular2-materialize#59]
**Setup:**
Angular: RC4
Module Loader/Bundler: Webpack

**Problem:**
I'm getting the following error message when loading my app:
> Couldn't find materialize function ''material_select' on element or the global Materialize object.

The error arises through this select component:
[Select component](https://gist.github.com/leNicDev/500f8533795bc25dd8eb355dc0cc5ba3
)

**Main file:**
[main.ts](https://gist.github.com/leNicDev/fac6c7e66e9a0343b0b67a8c8cd36aa3)

**Webpack config:**
[webpack.common.js](https://gist.github.com/leNicDev/117ad54f7aabbb8f28e108c151f398d1)

**Comment by rubyboy** [comment]
Thanks for the gists. I couldn't find a problem in them.
Does the webpack starter work for you? https://github.com/InfomediaLtd/angular2-webpack-starter
Also, can you check if you can access the global Waves object in the browser console after the app loaded? If you can then it means Materialize has loaded, but probably not in the right order (not before the angular2-materialize directive). If not, then it hasn't been loaded and it's a different issue.
If you could share your project I can try taking a look what else you be missing.

**Comment by leNicDev** [comment]
Sorry, I'm not allowed to share the project for different reasons.

But I found something out:
The Browser seems to know the global Materialize object since `Materialize.guid` is returning a function:

`> Materialize.guid`
returns
`function () {
	return s4() + s4() + '-' + s4() + '-' + s4() + '-' +
		s4() + '-' + s4() + s4() + s4();
}`

**Comment by rubyboy** [comment]
Did the webpack-starter work for you?
If it does, you can check differences to understand how it's different in its loading order.
I think it's something about Materialize being loaded too late. It needs to be loaded before the bootstrapping of your app, so the directive could access it when binding to the angular components.

**Comment by leNicDev** [comment]
The webpack-starter doesn't throw this error.
I don't see there any stylesheets that have to be required in HTML or the component itself. However, if I'm removing `<link href="../node_modules/materialize-css/dist/css/materialize.css" rel="stylesheet">` from my templates, not even the text inputs are working.

**Comment by rubyboy** [comment]
This is the bit that actually loads the CSS when materialize-css is required:
```
loaders: [
      {
        test: /materialize-css\/dist\/js\/materialize\.js/,
        loader: 'imports?materializecss'
      },
      //...
    ]
```

Here's the bit of code in the angular2-materialize that loads materialize and should kick off the loading of both JS and CSS: https://github.com/InfomediaLtd/angular2-materialize/blob/master/src/index.ts#L1

**Comment by omt66** [comment]
I have the same problem like @leNicDev, it cannot find the function and throws an exception. with the similar message as mentioned, in my case it is

> Error: Couldn't find materialize function ''tooltip' on element or the global Materialize object.

**Comment by rubyboy** [comment]
@omt66 I assume this is webpack as well? :)
Is that a project you can share? How different is it from the webpack starter? https://github.com/InfomediaLtd/angular2-webpack-starter
BTW, if you add a script import at the top of your page to the MaterializeCSS JS (right after jQuery), does it work then?

**Comment by omt66** [comment]
@rubyboy I tried adding Materializecss script import right after jQuery, and a few other combinations but none worked. It might be webpack issue. I was trying Angular RC4, I used the preboot's starter https://github.com/preboot/angular2-webpack , I just wanted to make sure that bare bone project with matrializecss would work but I was getting the similar error messages that people mentioned here in different bug reports.

**Comment by leNicDev** [comment]
Fixed the problem by downgrading jQuery 3.0.0 to 2.4.2.

**Comment by rubyboy** [comment]
Interesting. So MaterializeCSS doesn't work with jQuery 3.0.0... Here's the issue on the MaterialieCSS repo: https://github.com/Dogfalo/materialize/issues/3201
I wonder it that also explains the other Webpack issues people were having (with velocity)...
The MaterializeCSS page actually shows 2.1.1 as the recommended version in the getting started page (http://materializecss.com/getting-started.html), so I guess
